### PR TITLE
nye³

### PR DESCRIPTION
**Pull request by Joshinken**
i made it so that, if you use the analog battery detection, the code for the i2c chip isnt loaded in at all. i also changed it so that the banner update and banner draw functions dont each contain a copy of the code needed to draw the battery info and instead just call a separate function. recommend doing this to all other identical parts of the banner draw and banner update

### CODE REVIEW

**Comment by Joshinken** [comment]
i also wanted to add in an easier way to swap the logos, but i couldnt figure out where the "filename" even comes from, so i just didnt bother

**Comment by justcallmekoko** [comment]
You know you don't have to make a new pull request when you make changes? The pull request will automatically stay up to date with each commit you make in your branch.

I will test these changes a bit later today.

**Comment by Joshinken** [comment]
I DONT NEED TO MAKE NEW PULL REQUESTS

FUCK

i dont like github. it never tells you shit about fuck, you just need to already know how to use it. in order to bring my master up to date, i had to make a pull request to merge your master onto my master, and i feel thats not right, but its the only method i could find

**Comment by Joshinken** [comment]
yes, youll need to test if the i²c battery code still works as intended, as i dont have that chip, so it just doesnt draw a battery status.

**Comment by justcallmekoko** [comment]
I feel like if you plan on making more contributions to the project, I should probably send you an actual marauder.
At least then you would have the original hardware to test on.

**Comment by Joshinken** [comment]
that would be nice, yeah. i actually thought about buying a real marauder for testing purposes before, but i am SPECTACULARLY broke, thanks to the, uuuuuhhh... "local pandemic".

**Comment by justcallmekoko** [comment]
Shoot me an email and we can talk about getting you hooked up
at.the.red.team@gmail.com

**Comment by Joshinken** [comment]
holy shit thats a nice email.

**Comment by Joshinken** [comment]
give me a sec, i just noticed the thing is flickering again

**Comment by Joshinken** [comment]
the function i call to actually draw the battery status was taken from the draw_banner function, which doesnt update the battery_analog_last value, so it would continously be at 101, which is outside the valid range for battery_analog, which means they were never the same, which means it would always redraw the status as the values were always different.

tl;dr its a one-line fix

**Comment by Joshinken** [comment]
fixed it, also moved the float wd from the draw function, which saves 16 bytes and fixed the width issue

**Comment by Joshinken** [comment]
also im gonna massively overhault the drawing code for no reason other than "this still bothers me"

**Comment by Joshinken** [comment]
but thats still a while away

**Comment by Joshinken** [comment]
did i say "a while away"? cause i meant to say "done".

this was way easier than i thought and works way better

**Comment by Joshinken** [comment]
tiny efficency fix, had stuff like 
```
      for (n2 = wd; n2 >= -wd; n2--) {
        xlast2 = xlast + n2;
        x2 = x + n2;
        tft.drawLine(xlast2, ylast2, x2, y2, the_color);
        ylast2 = ylast + n;
        y2 = y + n;
        tft.drawLine(xlast2, ylast2, x2, y2, the_color);
      }
      }
```
where ylast2 and y2 kept being changed every loop, despite their values not actually changing

**Comment by justcallmekoko** [comment]
Hey just an FYI I tested the new battery changes with the original marauder hardware.
Changing the menu resulted in the battery percentage not being displayed on the status bar anymore because the battery percentage had not changed from the "old value". 

I made [these modifications](https://github.com/justcallmekoko/ESP32Marauder/pull/35/commits/9a51b8dc14610d5d880ef994af2fd498523eeed9) to get the status bar to update on menu change.

**Comment by Joshinken** [comment]
with your fix, does it still draw the battery status after changing menu several times?

**Comment by justcallmekoko** [comment]
Yes the battery status is drawn on the status bar as soon as the menu changes.

**Comment by Joshinken** [comment]
thanks! its a really simple bit of code once youve thought of it, but man did i not think of it at first.

right now i have nothing more to add and i havent found any more glitches, so i think this can be merged now.

**Comment by justcallmekoko** [comment]
Alright I will go ahead and merge this guy in
